Caltech-IPAC/firefly
# FIREFLY-1416: UI Conversion: Update landing page to have a polished look

Fixes [FIREFLY-1416](https://jira.ipac.caltech.edu/browse/FIREFLY-1416)

- [x] Make landing page as per [figma design](https://www.figma.com/file/YD5gkxtxfHhXiGWe7aSEBC/FF-Navigation?type=design&node-id=52%3A700&mode=design&t=PBDhUCQNbcf4gjsr-1)
- [x] Make landing page - empty results and app branding customisable
   - [x] refactor to slots design
   - [x] try building it with SUIT and adjust
- [x] Create a better way to show arrows pointing banner tabs and background tab
   - [x] Make such component using snackbar - AppHint (currently it's specific to landing page pointing banner and is not generalized enough because positioning it on app level like a tooltip is tricky)   
   - [x] Make it appear only first-time user is loading app
- [x] Polish tabs and banner
   - [x] Make the tab border extend to entire app to make it look everything below banner look like a tab panel
   - [x] Make dividers in banner more like in figma design
   - [x] Try ~shadow~, tab hover fix, make font shrink but not tab height on resize, and other tweaks
- [x] Polish side-menu/drawer

## Testing
Firefly: https://fireflydev.ipac.caltech.edu/firefly-1416-polish-landing/firefly/

IV: https://firefly-1416-polish-landing.irsakudev.ipac.caltech.edu/irsaviewer/

SUIT: https://suit-ui-conversion.irsakudev.ipac.caltech.edu/suit/

- Check landing page and notice hint about using tabs
- Do a long running search (I tried obscore searches) and send it to background to see background monitor hint
- Switch between dev and this version of apps to see minor (but important) differences in banner 

### New
- Check that both hints only appear first time app is loaded (note: you might need to clear local storage if you are doing repeated testing)
- Check drawer is following the design we finalized in slack
- Try resizing window, the tabs will shrink in width but not in height. Also hover of tabs is now same as Results.
- Check `LandingPage` component (renamed from DefaultLandingPage) being customized by IV [in this PR](https://github.com/IPAC-SW/irsa-ife/pull/310) and by SUIT [in this commit](https://github.com/lsst/suit/commit/8af10c20f79e0caaf697ce8d921f332680ca3bb8). 

## Early feedback needed on:
- Is landing page component customizable enough or does it require restructuring?
- Does AppHints look in appearance and functionality? (note: I have chosen solid primary to make them distinguishable from alerts and I think I have seen similar looking hints on some other Material based app on internet that I can't recall)
- The Empty results section doesn't occupy full height (unlike Figma design) - do we want that? Also, do we want to reduce spacing between inner elements (aka make the layout tighter) on smaller window sizes?
- Can I use browser local storage for storing one-time-show flag of app hints (similar to how theme is stored and remembered)?
- Tab hover background is bluish for Results tab and greyish for all other tabs - is it intentional? If not, I'd like to make it bluish for all other tabs because visually, it fits well with the soft primary background.
